### DISCUSSION TITLE
Hiragana/furigana mode for the Japanese.

### DISCUSSION THREAD

**Discussion by ganqqwerty** [openai/whisper#1650] [Ideas]
In Japanese, there are chinese characters (Kanji, e.g. 日本, "nihon, nippon") and there's an alphabet (Hiragana, e.g. にほん, "ni-ho-n"). Kanji has more than one pronunciation and depends on context. When it's not clear how to read a particular Kanji, they put Hiragana near the word like that: 日本(にほん). 

Whisper generates the kanji version, but it has all pronunciation information to generate hiragana version. In fact, it should be easier to generate hiragana because it's purely pronunciation, whereas there's multiple kanjis for a given pronunciation and the model have to make guesses.

Is it possible to implement the hiragana mode?

**Comment by jongwook** [comment]
Hi, this comment links to a notebook to fine-tune Whisper to transcribe in katakana-only:

- https://github.com/openai/whisper/discussions/64#discussioncomment-3765117

In theory, it could be possible to do something similar to below to suppress all kanji tokens to force Whisper to decode in hiragana, although the accuracy might be impacted:

- https://github.com/openai/whisper/discussions/1041#discussioncomment-5224837

**Reply by ganqqwerty** [reply]
supressing kanji sounds promising, will try!

**Comment by phineas-pta** [comment]
every language has its own quirk, nobody can implement a specific feature for each language, u have to do post-processing yourself

**Reply by dgoryeo** [reply]
Hi @ganqqwerty , this is an excellent suggestion. Assuming that Whipser does have that information, are there any ways to extract pronounciation information (or phonems)? @jongwook , would you help to shed some lights on this? Thanks.
